lib/cretonne/src/verifier.rs

```diff
@@ -185,56 +185,28 @@ impl<'a> Verifier<'a> {
 
     fn verify_entity_references(&self, inst: Inst) -> Result<()> {
         use ir::instructions::InstructionData::*;
+
+        for &arg in self.func.dfg[inst].arguments(&self.func.dfg.value_lists) {
+            self.verify_value(inst, arg)?;
+        }
```

Comment:
Yep. You can do the same thing with `second_result`. But you could also just let `DataFlowGraph::inst_results()` handle that and simply check the values returned by that iterator.

This means that you will get some assertion failures instead of the nicer verifier errors. I think that is OK if the fundamental data structures are corrupt: Linked lists of instruction results or EBB arguments.